src/examples/utils.ts

```diff
@@ -208,8 +208,15 @@ export function resolveNoBuildExample(
 }
 
 export function onHashChange(cb: () => void) {
-  window.addEventListener('hashchange', cb)
+  if (typeof document === 'undefined') return
+  const { pathname } = window.location
+  function handler() {
+    if (window.location.pathname === pathname) {
```

Comment:
Technically we could have also use .newURL / .oldURL on event argument, but looks like vitepress is dispatching only generic event object for hashchange. I'll fix it there and refactor this later.